db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/WindowResultSetNode.java

```diff
@@ -495,6 +495,35 @@ private void addOverColumns(Map<String, ColumnRefPosition> pulledUp) throws Stan
                         // Cache the ref
                         pulledUp.put(expKey, new ColumnRefPosition(bottomRC, childRCL.size()));
                     }
+                } else {
+                    // Check again if we find match, if not then add it in cache ref if we have the equivalent match
+                    if (pulledUp.get(expKey) == null) {
+                        ResultColumn srcRC = findOrPullUpRC(expRC,
+                                ((SingleChildResultSetNode) childResult).getChildResult(),
+                                rCtoCRfactory);
```

Comment:
This whole branch is exactly the same as the first half of " if (! contains(expRC, winRCL)) { ", could you consolidate it there?